napari/_qt/utils.py

```diff
@@ -302,3 +303,23 @@ def remove_flash_animation(widget: QWidget):
     """
     widget.setGraphicsEffect(None)
     del widget._flash_animation
+
+
+def connect_setattr(emitter, obj, attr):
```

Comment:
maybe we could put these in `event_utils.py`?  since they could potentially apply to both?
could also add a protocol type for emitter:

```python
if TYPE_CHECKING:
    class Emitter(Protocol):
        def connect(self, callback: Callable): ...
        def disconnect(self, callback: Callable): ...

def connect_setattr(emitter: Emitter, obj: object, attr: str): ...
```

Reply:
I cannot hide `Emitter` under `if TYPE_CHECKING:`

Reply:
why?

Reply:
(you can put it in quotes if you need to `emitter: 'Emitter'`)

Reply:
I forget about `from __future__ import annotations`.